url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/ScriptyEmbeddedRepl.java

```diff
@@ -0,0 +1,188 @@
+    {
+        // An infinite loop, waiting for client requests, spawning handlers and 
+        // wait for new requests.
+        //noinspection InfiniteLoopStatement
+        while (true)
+        {
+            try
+            {
+                // This call will block until a client contacts this server.
+                Socket socket = server.accept();
+                // Set the session timeout.
+                // After 30 min of inactivity the socket will be closed.
+                socket.setSoTimeout(1000 * 1800);
+
+                // We just got a connection from a client.
+                // We will create a separate handler for this client request (separate thread)
+                // and we will immediately keep listening for other clients to connect our server port.
+                // So this call starts the handling process, but it does not wait for it to finish.
+                new RequestHandler(socket, scripty, aUid, aPwd);
+            }
+            catch (IOException e)
+            {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+
+/**
+ * A request handler, it is created by the server.
+ * It will always run in a separate thread from the server so that the
+ * main server can keep listening for incoming connections.
+ *
+ */
+class RequestHandler
+implements Runnable
+{
+    private Socket socket;
+    private ReplEngine scripty;
+    private String uid;
+    private String pwd;
+
+    /*
+     * Create the handler, and start the processing of the request
+     * in a separate thread.
+     */
+    public RequestHandler(Socket aSocket, ReplEngine aEngine, String aUid, String aPwd)
+    {
+        this.socket = aSocket;
+        this.scripty = aEngine;
+        uid = aUid;
+        pwd = aPwd;
+        
+        Thread workerThread = new Thread(this);
+        workerThread.start();
+    }
+    
+    public boolean login(InputStream aIn, OutputStream aOut, String aUid, String aPwd)
+    {
+        try 
+        {
+            LineNumberReader input = new LineNumberReader(new InputStreamReader(aIn));
+            PrintWriter out = new PrintWriter(new OutputStreamWriter(aOut));
+            
+            int count = 0;
+            while(count++ < 5)
+            {
+                out.print("user: ");
+                out.flush();
+                String lUid = input.readLine();
+                
+                out.print("password: ");
+                out.flush();
+                String lPwd = input.readLine();
+
+                if(lUid != null && lUid.trim().equals(aUid) && lPwd != null && lPwd.trim().equals(aPwd)) return true;
+                else 
+                {
+                    out.println("Login failed.");
+                    out.flush();
+                }
+            }
+            return false;
+        } 
+        catch (Exception e) 
+        {
+            return false;
+        }
+    }
+
+    public void run()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 177-186 into a separate method.